packages/lisk-cryptography/test/nacl.js

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+// Require is used for stubbing
+const hash = require('../src/hash');
+const sign = require('../src/sign');
+const convert = require('../src/convert');
+const nacl = require('../src/nacl/nacl');
+
+const {
+	randombytes,
+	signKeyPair,
+	signDetached,
+	detachedVerify,
+	box,
+	boxOpen,
+} = nacl;
+
+const makeInvalid = str => {
+	const char = str[0] === '0' ? '1' : '0';
+	return `${char}${str.slice(1)}`;
+};
+
+describe('nacl', () => {
+	const CRYPTO_SIGN_PUBLICKEYBYTES = 32;
+	const CRYPTO_SIGN_PRIVATEKEYBYTES = 64;
+	const CRYPTO_SIGN_BYTES = 64;
+	const defaultPublicKey =
+		'7ef45cd525e95b7a86244bbd4eb4550914ad06301013958f4dd64d32ef7bc588';
+	const defaultPrivateKey =
+		'314852d7afb0d4c283692fef8a2cb40e30c7a5df2ed79994178c10ac168d6d977ef45cd525e95b7a86244bbd4eb4550914ad06301013958f4dd64d32ef7bc588';
+	const defaultMessage = 'Some default text.';
+	const defaultSignature =
+		'68937004b6720d7e1902ef05a577e6d9f9ab2756286b1f2ae918f8a0e5153c15e4f410916076f750b708f8979be2430e4cfc7ebb523ae1905d2ea1f5d24ce700';
+	const defaultEncryptedMessage =
+		'a232e5ea10e18249efc5a0aa8ed68271fc494d02245c52277ee2e14cddd960144a65';
+	const defaultNonce = 'df4c8b09e270d2cb3f7b3d53dfa8a6f3441ad3b14a13fb66';
+
+	let hashStub;
+	let digestMessageStub;
+	let convertPublicKeyEd2CurveStub;
+	let convertPrivateKeyEd2CurveStub;
+
+	beforeEach(() => {
+		hashStub = sandbox
+			.stub(hash, 'default')
+			.returns(
+				Buffer.from(
+					'314852d7afb0d4c283692fef8a2cb40e30c7a5df2ed79994178c10ac168d6d97',
```

Comment:
It's a little confusing that this is also the start of the default hash. Prefer totally separate data.